pkg/realizer/deliverable/realizer.go

```diff
@@ -105,7 +105,34 @@ func generateRealizedResource(resource v1alpha1.DeliveryResource, template templ
 			APIVersion: v1alpha1.SchemeGroupVersion.String(),
 		}
 
-		outputs = template.GenerateResourceOutput(output)
+		var err error
+		outputs, err = template.GenerateResourceOutput(output)
+		if err != nil {
+			for _, previousResource := range previousResources {
+				if previousResource.Name == resource.Name {
+					outputs = previousResource.Outputs
+					break
+				}
+			}
+		} else {
+			currTime := metav1.NewTime(time.Now())
+			for j, out := range outputs {
+				outputs[j].LastTransitionTime = currTime
```

Comment:
do you think we could split some of the following one/more functions? 😅   this code code path is getting tough to follow